File: Rust/Daily002_easy/InterestCalculator.rs
// INTEREST CALCULATOR
//
// challenge #2 (easy)
// http://www.reddit.com/r/dailyprogrammer/comments/pjbj8/easy_challenge_2/
//
// sarthak7u@gmail.com
//

fn main() {
    // simple interest
    println!("{}",simple_interest(100.0, 5.0, 10.0));

    // comppund interest
    println!("{}",compound_interest(100.0, 5.0, 10.0));
}

fn simple_interest(principal: f64, rate: f64, time: f64) -> f64 {
  return principal * rate * time / 100.0;
}

fn compound_interest(principal: f64, rate: f64, time: f64) -> f64 {
  return principal * (1.0 + rate / 100.0).powf(time) - principal;
}

File: OCaml/Daily002_easy/interestCalculator.ml
(*
  Compound interest calculator
  Challenge #2 (easy)
  https://www.reddit.com/r/dailyprogrammer/comments/pjbj8/easy_challenge_2/

  sarthak7u@gmail.com
*)

let compound_interest p r n t =
  p *. Float.pow (1.0 +. (r /. n)) (n *. t)

let main () = 
  (* let f = compound_interest 10000.0 0.06 12.0 20.0 in *)
  let f = compound_interest 10000.0 0.07 12.0 25.0 in

  print_float f;
  print_newline ()

let () = main()
